expertisesolutions/efl
# windows: Add strings.h functions to work with ucrt on windows

# Goal
This aims to solve strings.h dependency from posix, to work with native Universal C Runtime of windows.

# How it work
From posix strings.h functions, we have: 
- ffs
- strcasecmp
- strncasecmp
- strcasecmp_l
- strncasecmp_l

In case of `ffs`, i've made a dumb implementation located in 'src\lib\evil\unposix\strings.c'.

In all remaining functions, i did an #define `<posix_function>` to `_<ucrt_functions>`

# how do I test it
You can configure and build this branch on windows with: .\configure && .\build -k0, there isn't should be errors, with related functions.

I made too a .c code that use theses functions. These code are located in src\tests\evil\unposix\test_evil_unposix_strings.c. 
To run it, you need execute ./run_test.sh int git bash.
